### PR TITLE
miner: retrieve mining state from live database

### PR DESCRIPTION
**Pull request by rjl493456442**
This PR drops the necessity of recovering state for mining block.

In PoW, the state of head block is always used as base for preparing a mining block. It's unnecessary to recover state at all.
In PoS, the mining block is always generated with a `ForkchoiceUpdatedV1` request from consensus client with the head block specified. It means the base state should always be prepared for new mining block.

All in all, it's unnecessary to have this complicated mechanism for recovering a state for new mining block.

### CODE REVIEW

**Comment by MariusVanDerWijden** [comment]
We check and ignore fcu's to old but canonical payloads in the api. So stale fcu's should be filtered out

**Comment by rjl493456442** [comment]
> We check and ignore fcu's to old but canonical payloads in the api. So stale fcu's should be filtered out

Where is it? It's in CL or EL? If in EL could please share the code link?

File: eth/catalyst/api.go [hunk 1]
```diff
@@ -140,16 +140,26 @@ func (api *ConsensusAPI) ForkchoiceUpdatedV1(update beacon.ForkchoiceStateV1, pa
...
 		}
+	} else if api.eth.BlockChain().CurrentBlock().Hash() == update.HeadBlockHash {
+		// If the specified head matches with our local head, do nothing and keep
+		// generating the payload. It's a special corner case that a few slots are
+		// missing and we are requested to generate the payload in slot.
 	} else {
 		// If the head block is already in our canonical chain, the beacon client is
 		// probably resyncing. Ignore the update.
 		log.Info("Ignoring beacon update to old head", "number", block.NumberU64(), "hash", update.HeadBlockHash, "age", common.PrettyAge(time.Unix(int64(block.Time()), 0)), "have", api.eth.BlockChain().CurrentBlock().NumberU64())
+		return valid(nil), nil
```
[the remark this answers is not in the document]
Comment (MariusVanDerWijden) [reply eth/catalyst/api.go:162]: Yeah that was what we discussed today at stabby. I wouldn't like to make any decisions based on the finalization point.

**Review by karalabe — APPROVED** [review]
SGTM

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**